dnn_rotate/nodes/rotate.py

```diff
@@ -0,0 +1,154 @@
+
+# Utility function to convert radians to degrees
+def degrees(r):
+    return 180.0 * r / math.pi
+
+
+# Class to encapsulate our node
+class Rotate:
+    """
+    Constructor for our class
+    """
+    def __init__(self):
+        self.ok = False
+        rospy.init_node("rotate")
+
+        # These paramaters allow us to caluclate the angle per pixel in the image
+        # It could alternatively be done with camera_info topic
+        self.image_width = rospy.get_param("~image_width", 410)
+        self.field_of_view = rospy.get_param("~field_of_view", 0.925)
+
+        # Setup the rotate service we serve
+        self.srv = rospy.Service("rotate", dnn_rotate.srv.String,
+                                 self.service_callback)
+        
+        # Create a proxy object for the move action server
+        self.move = actionlib.SimpleActionClient('move_base',
+                                                 move_base_msgs.msg.MoveBaseAction)
+        rospy.loginfo("Waiting for move service to be available")
+        if not self.move.wait_for_server(rospy.Duration(10.0)):
+           rospy.logerr("Move service not available")
+           return
+
+        # Create a proxy object fot the trigger service we call
+        rospy.loginfo("Waiting for detect service to be available")
+        try:
+            rospy.wait_for_service("/dnn_detect/detect", 10)
+            self.trigger = rospy.ServiceProxy("/dnn_detect/detect",
+                                              dnn_detect.srv.Detect)
+            rospy.loginfo("Rotate node ready")
+            self.ok = True
+        except:
+            rospy.logerr("detect service is not available") 
+
+    # This is called when we receive a rotate service call
+    def service_callback(self, rotate_req):
+        target = rotate_req.object.data
+        print("Received service call: rotate %s" % target)
+
+        # Create a response to our service
+        response = dnn_rotate.srv.StringResponse()
+
+        # Make a detection trigger service call
+        detect_req = dnn_detect.srv.DetectRequest()
+        detect_resp = self.trigger(detect_req)
+
+        # Look for matching objects
+        best_target = None
+        best_confidence = 0.0
+        for object in detect_resp.result.objects:
+            if object.class_name == target:
+                found_target = True
+                if object.confidence > best_confidence:
+                    best_target = object
+                    best_confidence = object.confidence
+
+        # Create a negative response to our rotate service call
+        if best_target is None: 
+            response.response.data = "No %s object found" % target
+            return response
+
+        # Calculate how far we need to rotate
+        x_center = (best_target.x_min + best_target.x_max) / 2.0
+        angle_per_pixel = self.field_of_view / self.image_width
+        angle_to_rotate = (self.image_width/2.0 - x_center) * angle_per_pixel
```

Comment:
*Technically* the angle here needs to transformed into base_link, in case of any weird camera mount positions. Upside down, sideways etc.